modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -1545,8 +1546,10 @@ void simplifySubgraphs(opencv_onnx::GraphProto& net)
     subgraphs.push_back(makePtr<MishSubgraph>());
     subgraphs.push_back(makePtr<NormalizeSubgraph4>());
     subgraphs.push_back(makePtr<NormalizeSubgraph5>());
+#if FUSION_DNN_BACKEND_OPENCV
```

Comment:
Why not compare `OPENCV_DNN_BACKEND_DEFAULT` with `DNN_BACKEND_INFERENCE_ENGINE` in runtime?

Reply:
I actually want to shut it down when it is not DNN_BACKEND_OPENCV.

Reply:
Yes, but this is the same but we don’t need add a definition nut check a value

Reply:
`OPENCV_DNN_BACKEND_DEFAULT` could be undef during compilation, so cannot always compare `OPENCV_DNN_BACKEND_DEFAULT` with `DNN_BACKEND_INFERENCE_ENGINE` in runtime.

Reply:
@fengyuentau, I mean something like that:
```cpp 
  if (getParam_DNN_BACKEND_DEFAULT()== DNN_BACKEND_INFERENCE_ENGINE)
  {
    subgraphs.push_back(makePtr<AttentionSubGraph>());
    subgraphs.push_back(makePtr<AttentionSingleHeadSubGraph>());
  }
#endif

Reply:
I guess it should be `if (getParam_DNN_BACKEND_DEFAULT()== DNN_BACKEND_OPENCV)` instead right?

Reply:
Tests about Attention fusion in graph simplifier cannot be in sync with this.

Reply:
This is because https://github.com/opencv/opencv/pull/24672 has been merged but https://github.com/opencv/opencv/pull/24476/files#r1421301615 not applied. Can you do so?

Reply:
Yes, but also I can say that `PR:4.x / Ubuntu2004-x64-OpenVINO / BuildAndTest` will still fail to pass because the attention subgraph fusion is turned off for openvino but the tests of graph simplifier do not.

Reply:
If only OpenVINO is configured as a default backend in CMake flags. On our CI there is no such option: https://github.com/opencv/ci-gha-workflow/blob/49062b85612df5203c48735fdb107ffeff4cd800/.github/workflows/OCV-PR-4.x-U20-OpenVINO.yaml#L16. So test should pass.